theano/tests/test_breakpoint.py

```diff
@@ -0,0 +1,71 @@
+import numpy
+import theano
+import theano.tensor as T
+from theano.tests.breakpoint import PdbBreakpoint
+
+class TestPdbBreakpoint:
+
+    def setup(self):
+
+        # Sample computation that involves tensors with different numbers
+        # of dimensions
+        self.input1 = T.fmatrix()
+        self.input2 = T.fscalar()
+        self.output = T.dot((self.input1 - self.input2),
+                            (self.input1 - self.input2).transpose())
+
+        # Declare the conditional breakpoint
+        self.breakpointOp = PdbBreakpoint("Sum of output too high")
+        self.condition = T.gt(self.output.sum(), 1000)
+        (self.monitored_input1,
+         self.monitored_input2,
+         self.monitored_output) = self.breakpointOp(self.condition,
+                                                    self.input1,
+                                                    self.input2, self.output)
+
+    def test_infer_shape(self):
+
+        input1_value = numpy.arange(9).reshape(3,3).astype("float32")
```

Comment:
When testing for shapes, do not use square shapes. This lower the test quality. We could mix the dimensions and the tests won't see it. Also, we have function to tests the infer_shape better then what is done it. That would be better.